doc/source/reference/arrays.scalars.rst

```diff
@@ -169,6 +174,25 @@ Inexact types
    :members: __init__
    :exclude-members: __init__
 
+.. note::
+
+   The printing behavior of inexact scalars is to use as few digits as possible
+   to display the result unambiguously. This means that equal values at
+   different precisions may display differently::
```

Comment:
It could be nice to point the reader to the docstrings of `np.set_printoptions`, and `np.format_float_positional/np.format_float_scientific` for descriptions of the "unique" printing option.

Maybe we can even re-use the same language, as in:

 >  Inexact scalars are printed using the fewest decimal digits needed to uniquely identify the inexact number from other values of the same datatype, by judicious rounding. See the `unique` parameter of `np.format_float_positional` and `np.format_float_scientific`, and the `floatmode='unique' option of `np.set_printoptions` for more information.  This means that variables with equal binary values but whose datatypes are of different precisions may display differently::